### PR TITLE
Some miscellaneous fixes

### PR DESCRIPTION
**Pull request by josephsutton1**
Just a collection of minor fixes that I had lying around.

### CODE REVIEW

File: kdc/fast.c [hunk 1]
```diff
@@ -799,8 +799,10 @@ _kdc_fast_strengthen_reply_key(astgs_request_t r)
 
 	ret = krb5_generate_random_keyblock(r->context, r->reply_key.keytype,
 					    &r->strengthen_key);
-	if (ret)
-	    krb5_abortx(r->context, "random generator fail");
+	if (ret) {
+	    kdc_log(r->context, r->config, 0, "random generator fail");
+	    return ret;
+	}
```
Comment (abartlet) [inline kdc/fast.c:805]: Should we say why it failed - given the reason we are not abort()ing is that it can fail for reasons unrelated to the RNG?

**Review by abartlet — COMMENTED** [review]
Other than clarifying further why the RNG failed after stopping it aborting, I'm happy with these.

File: lib/krb5/changepw.c [hunk 2]
```diff
@@ -695,7 +695,7 @@ krb5_change_password (krb5_context	context,
 		      int		*result_code,
 		      krb5_data		*result_code_string,
 		      krb5_data		*result_string)
-    KRB5_DEPRECATED_FUNCTION("Use X instead")
+    KRB5_DEPRECATED_FUNCTION("Use krb5_set_password instead")
```
Comment (abartlet) [inline lib/krb5/changepw.c:698]: I've carefully checked all the Use X instead changes match the comments.

**Review by abartlet — APPROVED** [review]

File: kdc/fast.c [hunk 3]
```diff
@@ -799,8 +799,10 @@ _kdc_fast_strengthen_reply_key(astgs_request_t r)
 
 	ret = krb5_generate_random_keyblock(r->context, r->reply_key.keytype,
 					    &r->strengthen_key);
-	if (ret)
-	    krb5_abortx(r->context, "random generator fail");
```
Comment (nicowilliams) [inline kdc/fast.c:803]: I'm of the opinion that the KDC should definitely `abort()` if there's no RNG.  The RNG shouldn't fail though.

File: lib/base/dict.c [hunk 4]
```diff
@@ -52,8 +52,10 @@ dict_dealloc(void *ptr)
 {
     heim_dict_t dict = ptr;
     struct hashentry **h, *g, *i;
+    size_t j;
 
-    for (h = dict->tab; h < &dict->tab[dict->size]; ++h) {
```
Comment (nicowilliams) [inline lib/base/dict.c:56]: `&dict->tab[dict->size]` is perfectly safe and defined behavior.

File: kdc/fast.c (hunk 3 above)
Comment (nicowilliams) [reply kdc/fast.c:803]: `krb5_generate_random_block()` does `krb5_abortx()` if `krb5_generate_random()` fails, and `krb5_generate_random_keyblock()` calls `krb5_generate_random_block()`.  We should rationalize this a bit.

Comment (nicowilliams) [reply kdc/fast.c:803]: `krb5_generate_random_block()` is `void`-returning.  We should deprecate it and not use it in-tree.

**Comment by nicowilliams** [comment]
I think [f3ad53c1d](https://github.com/heimdal/heimdal/pull/1159/commits/f3ad53c1d125372da265685f715ba8990dbe30aa) is not needed but not harmful, so I'm inclined to just take it, though if you have time to remove that commit that'd be great.

**Comment by josephsutton1** [comment]
> I think [f3ad53c1d](https://github.com/heimdal/heimdal/pull/1159/commits/f3ad53c1d125372da265685f715ba8990dbe30aa) is not needed but not harmful, so I'm inclined to just take it, though if you have time to remove that commit that'd be great.

OK, removed.

**Comment by nicowilliams** [comment]
Thanks!

**Pull request merged**